subsys/bluetooth/services/hids_c.c

```diff
@@ -941,25 +941,58 @@ int bt_gatt_hids_c_rep_write(struct bt_gatt_hids_c *hids_c,
 	return 0;
 }
 
+/**
+ *  @brief Process report write without response
+ *
+ *  @param conn Connection object.
+ *  @param rep_ptr Report object.
+ */
+static void rep_write_wo_rsp_process(struct bt_conn *conn, void *rep_ptr)
+{
+	struct bt_gatt_hids_c_rep_info *rep = rep_ptr;
+
+	if (!rep->write_cb) {
+		LOG_ERR("No write callback present");
+		return;
+	}
+
+	rep->write_cb(rep->hids_c, rep, 0);
```

Comment:
Is error parameter realy needed in this callback since it is always set to 0 (no error) ? It might be confusing for user.

Reply:
I wanted to use common API and single callback for write and write without response to keep things simpler. Do you think we need to make separate callback here?

Reply:
I think you can use a common callback. For me you do not have to create a new callback only for this operation.